R-package/NAMESPACE

```diff
@@ -52,6 +53,7 @@ export(xgb.plot.tree)
 export(xgb.save)
 export(xgb.save.raw)
 export(xgb.serialize)
+export(xgb.set.config)
```

Comment:
Since you're adding a new dependency, `{jsonlite}`, I expect it to show up in the diff on NAMESPACE.

I think you need to add `importFrom`s to this list: https://github.com/dmlc/xgboost/blob/master/R-package/R/xgboost.R#L78

```r
#' @importFrom jsonlite fromJSON
#' @importFrom toJSON
```

As a side note, I have a personal preference for putting these `importFrom` comments on the code that uses the imports, not grouping them all together in one file. Makes it harder to make mistakes like this, and it means that when code is removed from your package, the imports it carried with it are automatically removed too, without you needing to remember to go update a separate file. But that's bigger than this PR.

Reply:
@jameslamb I read that `@importFrom` is not necessary when `::` operator is used to access individual functions: https://kbroman.org/pkg_primer/pages/depends.html

Reply:
It's something that R people argue about haha.

I personally like to use both. I think of it like this:

* `::` = guarantees the correctness of your code (and makes it slightly faster because R doesn't search for symbols across all attached namespaces)
* `importFrom` = documentation that makes the `NAMESPACE` file readable as a source of truth for *"what are all the packages this package depends on, and which specific objects does it need?"*

Reply:
Got it. Let me add `importForm` for jsonlite functions.